charset/charset.go

```diff
@@ -421,6 +431,7 @@ var collations = []*Collation{
 	{246, "utf8mb4", "utf8mb4_unicode_520_ci", false},
 	{247, "utf8mb4", "utf8mb4_vietnamese_ci", false},
 	{255, "utf8mb4", "utf8mb4_0900_ai_ci", false},
+	{2048, "utf8mb4", "utf8mb4_tidb_zh_pinyin_cs", false},
```

Comment:
```suggestion
	{2048, "utf8mb4", "utf8mb4_zh_tidb_pinyin_cs", false},
```

Reply:
> are we sure we want case-sensitive

yes, cs implemention will be eaiser, and the newest chinese language collation `utf8mb4_zh_0900_cs_as` is cs too